communication/vehicleconnections/mavsdkvehicleconnection.cpp

```diff
@@ -14,28 +14,106 @@ MavsdkVehicleConnection::MavsdkVehicleConnection(std::shared_ptr<mavsdk::System>
     mVehicleType = vehicleType;
     mMavlinkPassthrough.reset(new mavsdk::MavlinkPassthrough(system));
 
-    // Setup gimbal
-    mComponentDiscoveredHandle = mSystem->subscribe_component_discovered([this](mavsdk::System::ComponentType){
-        if (mSystem->has_gimbal() && mGimbal.isNull()) {
-            mGimbal = QSharedPointer<MavsdkGimbal>::create(mSystem);
-            emit detectedGimbal(mGimbal);
-        }
-    });
-    if (mSystem->has_gimbal() && mGimbal.isNull()) // Gimbal might have been discovered before callback was registered
-        mGimbal = QSharedPointer<MavsdkGimbal>::create(mSystem);
+    // Set up param plugin
+    mParam.reset(new mavsdk::Param(mSystem));
 
     switch (mVehicleType) {
     case MAV_TYPE_QUADROTOR:
         qDebug() << "MavsdkVehicleConnection: we are talking to a MAV_TYPE_QUADROTOR / PX4.";
         mVehicleState = QSharedPointer<CopterState>::create(mSystem->get_system_id());
         mVehicleState->setName("Copter " + QString::number(mSystem->get_system_id()));
+
+        // Setup gimbal
+        mComponentDiscoveredHandle = mSystem->subscribe_component_discovered([this](mavsdk::System::ComponentType component_type){
+            if (component_type == mavsdk::System::ComponentType::GIMBAL && mSystem->has_gimbal() && mGimbal.isNull()) {
+                mGimbal = QSharedPointer<MavsdkGimbal>::create(mSystem);
+                emit detectedGimbal(mGimbal);
+            }
+        });
+        if (mSystem->has_gimbal() && mGimbal.isNull()) // Gimbal might have been discovered before callback was registered
+            mGimbal = QSharedPointer<MavsdkGimbal>::create(mSystem);
+
         break;
     case MAV_TYPE_GROUND_ROVER:
-        qDebug() << "MavsdkVehicleConnection: we are talking to a MAV_TYPE_GROUND_ROVER / WayWise.";
-        // TODO: detect different rover types...
-        mVehicleState = QSharedPointer<CarState>::create(mSystem->get_system_id());
-        mVehicleState->setName("Car " + QString::number(mSystem->get_system_id()));
-        break;
+        {
+            auto mWaywiseObjectType = WAYWISE_OBJECT_TYPE_CAR;
+            auto waywiseObjTypeParamResult = getIntParameterFromVehicle("VEH_WW_OBJ_TYPE");
+            if (waywiseObjTypeParamResult.first == VehicleConnection::Result::Success) {
+                mWaywiseObjectType = static_cast<WAYWISE_OBJECT_TYPE>(waywiseObjTypeParamResult.second);
+                if (!(mWaywiseObjectType == WAYWISE_OBJECT_TYPE_TRUCK || mWaywiseObjectType == WAYWISE_OBJECT_TYPE_CAR)){
```

Comment:
IMO, checking for mWaywiseObjectType  should be done in a switch case, where you can combine the WAYWISE_OBJECT_TYPE_CAR and default case like:
switch (mWaywiseObjectType) {
WAYWISE_OBJECT_TYPE_CAR:
default:
    ...;
   break;
}